src/stores/registerChallenge.ts

```diff
@@ -182,6 +196,123 @@ export const useRegisterChallengeStore = defineStore('registerChallenge', {
     setMerchandiseCards(cards: Record<Gender, MerchandiseCard[]>) {
       this.merchandiseCards = cards;
     },
+    /**
+     * Load registration data from API and set store state
+     */
+    async loadRegisterChallengeToStore(): Promise<void> {
+      const { registrations, loadRegistrations } = useApiGetRegisterChallenge(
+        this.$log,
+      );
+      this.isLoadingRegisterChallenge = true;
+      await loadRegistrations();
+      if (registrations.value.length > 0) {
+        this.setRegisterChallengeFromApi(registrations.value[0]);
+      } else {
+        this.$log?.debug('No registration data available to set store state');
+      }
+      this.isLoadingRegisterChallenge = false;
+    },
+    /**
+     * Set store state from API registration data
+     * @param registration - Registration data from API
+     */
+    setRegisterChallengeFromApi(registration: RegisterChallengeResult): void {
+      this.$log?.debug(
+        `Setting store state from registration data <${JSON.stringify(
+          registration,
+          null,
+          2,
+        )}>`,
+      );
+      const storeData = registerChallengeAdapter.toStoreData(registration);
+      // update store state
+      this.setPersonalDetails(storeData.personalDetails);
+      this.$log?.debug(
+        `Personal details updated to <${JSON.stringify(this.personalDetails, null, 2)}>`,
+      );
+      // ! if subject = 'company' or 'school', amount corresponds to donation
+      // TODO: load payment amount from API into the payment step UI
+      // this.setPaymentAmount(storeData.paymentAmount);
+      // this.$log?.debug(
+      //   `Payment amount updated to <${this.paymentAmount}>`,
+      // );
+      // TODO: set voucher name to display on complete payment step
+      this.setPaymentSubject(storeData.paymentSubject);
+      this.$log?.debug(`Payment subject updated to <${this.paymentSubject}>`);
+      // preset organization type based on payment subject
+      if (storeData.paymentSubject === PaymentSubject.company) {
+        this.setOrganizationType(OrganizationType.company);
+      } else if (storeData.paymentSubject === PaymentSubject.school) {
+        this.setOrganizationType(OrganizationType.school);
+      } else {
+        // if payment subject is "individual" or "voucher"
+        // ! resetting type will reset organization, subsidiary and team IDs
```

Comment:
@tmszi the UI requires to set `organizationType` here to enable loading appropriate options for organization, subsidiary and team. Otherwise, UI action (manually selecting type in the "participation" step) resets the organization, subsidiary and team IDs.
Could we add it as a field in the GET response so that it can be set here?

Reply:
Implemented.